packages/xo-web/src/common/xo/migrate-vdi-modal/index.js

```diff
@@ -0,0 +1,73 @@
+import _ from 'intl'
+import Component from 'base-component'
+import Icon from 'icon'
+import PropTypes from 'prop-types'
+import React from 'react'
+import SingleLineRow from 'single-line-row'
+import { Container, Col } from 'grid'
+import { createCompare } from 'utils'
+import { createSelector } from 'selectors'
+import { SelectSr } from 'select-objects'
+
+import { isSrShared } from '../'
+
+const createCompareContainers = poolId =>
+  createCompare([_ => _.$pool === poolId, _ => _.type === 'pool'])
+const compareSrs = createCompare([isSrShared])
+
+export default class MigrateVdiModalBody extends Component {
+  static propTypes = {
+    checkSr: PropTypes.func.isRequired,
+    pool: PropTypes.string.isRequired,
+  }
+
+  get value() {
+    return this.state
+  }
+
+  _getCompareContainers = createSelector(
+    () => this.props.pool,
+    createCompareContainers
+  )
+
+  _checkSr = createSelector(
+    () => this.props.checkSr,
+    () => this.state.sr,
+    (check, sr) => check(sr)
+  )
+
+  render() {
+    return (
+      <Container>
+        <SingleLineRow>
+          <Col size={6}>{_('vdiMigrateSelectSr')}</Col>
+          <Col size={6}>
+            <SelectSr
+              compareContainers={this._getCompareContainers()}
+              compareOptions={compareSrs}
+              onChange={this.linkState('sr')}
+              required
+            />
+          </Col>
+        </SingleLineRow>
+        <SingleLineRow className='mt-1'>
+          <Col>
+            <label>
+              <input type='checkbox' onChange={this.linkState('migrateAll')} />{' '}
+              {_('vdiMigrateAll')}
```

Comment:
So what's the difference between this and just selecting all the VDIs in the table and migrating them?

Reply:
If we select all VDIs in the table, the snapshots and the base copies will also be selected, so we will not be able to migrate them.

Reply:
Ok I understand. But I wonder if there's a risk that a user would think that it does something else and accidentally migrates all the SR's VDIs. He could still easily do that by selecting "Normal disks" in the SortedTable, right?

Reply:
Yes,  but it already exists and no feedback from the client for this. 
I see that it's clear because there is a title `Migrate all VDIs`.

Reply:
It already exists on a VM with very few disks compared to a SR. It can wait another PR but I think we should remove it:
- it's "dangerous" on SRs since there isn't an extra confirm for this
- it doesn't really make sense to go through a "migrate some VDIs" modal to migrate them all
- you can easily do this by filtering the VDIs in the table